### PR TITLE
ref: Auto-fix some ruff ANN rules

### PR DESCRIPTION
**Pull request by cbornet**
Auto-fix some ruff ANN rules

### CODE REVIEW

File: src/backend/base/langflow/custom/custom_component/component.py [hunk 1]
```diff
@@ -311,7 +311,7 @@ def validate(self, params: dict) -> None:
 
     def _set_output_types(self) -> None:
         for output in self._outputs_map.values():
-            return_types = self._get_method_return_type(output.method)
+            return_types = self._get_method_return_type(output.method or "")
```
Comment (cbornet) [inline src/backend/base/langflow/custom/custom_component/component.py:314]: `output.method` can be None. 
Is it OK or should an exception be raised ?

File: src/backend/base/langflow/custom/custom_component/component.py [hunk 2]
```diff
@@ -326,8 +326,7 @@ def _set_output_required_inputs(self) -> None:
                 source_code = inspect.getsource(method)
                 ast_tree = ast.parse(dedent(source_code))
             except Exception:  # noqa: BLE001
-                source_code = self._code
-                ast_tree = ast.parse(dedent(source_code))
+                ast_tree = ast.parse(dedent(self._code or ""))
```
Comment (cbornet) [inline src/backend/base/langflow/custom/custom_component/component.py:329]: `self._code` is not None by construction

File: src/backend/base/langflow/graph/graph/state_manager.py [hunk 3]
```diff
@@ -33,15 +33,3 @@ def get_state(self, key, run_id: str):
 
     def subscribe(self, key, observer: Callable) -> None:
         self.state_service.subscribe(key, observer)
-
-    def notify_observers(self, key, new_state) -> None:
```
Comment (cbornet) [inline src/backend/base/langflow/graph/graph/state_manager.py]: not used and incorrect (self.observers doesn't exist)

File: src/backend/base/langflow/graph/schema.py [hunk 4]
```diff
@@ -57,14 +57,6 @@ class InterfaceComponentTypes(str, Enum, metaclass=ContainsEnumMeta):
     DataOutput = "DataOutput"
     WebhookInput = "Webhook"
 
-    def __contains__(cls, item) -> bool:
```
Comment (cbornet) [inline src/backend/base/langflow/graph/schema.py]: inherited from ContainsEnumMeta

File: src/backend/base/langflow/graph/vertex/types.py [hunk 5]
```diff
@@ -366,6 +366,7 @@ async def _run(self, *args, **kwargs) -> None:
             self._built_result = self._built_object
 
         else:
+            # TODO: _run doesn't exist in ComponentVertex
             await super()._run(*args, **kwargs)
```
Comment (cbornet) [inline src/backend/base/langflow/graph/vertex/types.py:370]: `ComponentVertex` doesn't have a `_run` method.
Not sure what to do here: ignore or raise exception ?

File: src/backend/base/langflow/services/cache/service.py [hunk 6]
```diff
@@ -245,6 +247,10 @@ async def set(self, key, value, lock=None) -> None:
             msg = "RedisCache only accepts values that can be pickled. "
             raise TypeError(msg) from exc
 
+    @override
+    async def set(self, key, value, lock=None) -> None:
+        self._set(key, value)
```
Comment (cbornet) [inline src/backend/base/langflow/services/cache/service.py:252]: Calling blocking method here 🤔.
Should use the `redis.asyncio` client ?

File: src/backend/base/langflow/custom/custom_component/component.py (hunk 1 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/custom/custom_component/component.py:314]: I think an exception is a good idea.

File: src/backend/base/langflow/custom/custom_component/component.py (hunk 2 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/custom/custom_component/component.py:329]: LGTM

File: src/backend/base/langflow/graph/graph/state_manager.py (hunk 3 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/graph/graph/state_manager.py]: LGTM

File: src/backend/base/langflow/services/cache/service.py (hunk 6 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/services/cache/service.py:252]: Sure

**Review by ogabrielluiz — APPROVED** [review]
LGTM

File: src/backend/base/langflow/graph/vertex/types.py (hunk 5 above)
Comment (cbornet) [reply src/backend/base/langflow/graph/vertex/types.py:370]: Not sure how to test this either as `self.is_interface_component` is always true for InterfaceVertex.
So this looks like dead code.
Maybe we can just remove the if/else ?

File: src/backend/base/langflow/custom/custom_component/component.py (hunk 1 above)
Comment (cbornet) [reply src/backend/base/langflow/custom/custom_component/component.py]: done.PTAL.

File: src/backend/base/langflow/graph/vertex/types.py (hunk 5 above)
Comment (cbornet) [reply src/backend/base/langflow/graph/vertex/types.py]: I removed the if/else. PTAL.

**Pull request merged**